app/routes/send-endringsmelding.tsx

```diff
@@ -2,119 +2,142 @@ import HovedInnhold from '~/komponenter/hovedInnhold/HovedInnhold';
+  useSpråk,
+  useTekster,
+} from '~/hooks/contextHooks';
 import { Button, Textarea } from '@navikt/ds-react';
-import { useNavigate } from '@remix-run/react';
+import { Form, useActionData, useNavigate, useSubmit } from '@remix-run/react';
 import Veiledning from '~/komponenter/veiledning/Veiledning';
 import css from './send-endringsmelding.module.css';
 import { ETypografiTyper } from '~/typer/typografi';
 import { ESanityMappe, ESteg } from '~/typer/felles';
 import { hentPathForSteg } from '~/utils/hentPathForSteg';
+import { sendEndringsmelding } from '~/utils/sendEndringsmelding';
+import { ActionArgs } from '@remix-run/node';
+import { EFritekstFeil } from '~/typer/fritekstfeil';
+import {
+  MAKS_INPUT_LENGDE,
+  RESPONSE_STATUS_OK,
+  SPESIAL_TEGN_REGEX,
+} from '~/konstanter/sendEndringsmelding';
+import { getSession } from '~/sessions';
+import { i18nInnhold } from '~/utils/i18n';
+
+export async function action({ request }: ActionArgs) {
+  const formData = await request.formData();
+  const endringsmelding = formData.get('endringsmelding') as string;
+  return await sendEndringsmelding(
+    endringsmelding,
+    await getSession(request.headers.get('Cookie')),
+  ).then(response => {
+    if (!response.ok) {
+      console.log('Feil i send endringsmelding');
+      throw Error('Det skjedde en feil under POST til backend');
+    }
+    return response;
+  });
+}
 
 export default function SendEndringsmelding() {
+  const actionData = useActionData<typeof action>();
+  const submit = useSubmit();
+
   const tekster = useTekster(ESanityMappe.SEND_ENDRINGER);
   const teksterFelles = useTekster(ESanityMappe.FELLES);
-
   const navigate = useNavigate();
   const [språk] = useSpråk();
+  const [, settEndringsmeldingMottattDato] = useEndringsmeldingMottattDato();
 
-  const spesialTegnRegex = /[!@#$%^&*()?"{}|<>+¨=]/;
-  const MAKS_INPUT_LENGDE = 1000;
+  const [valideringsfeil, settValideringsfeil] = useState<EFritekstFeil | null>(
+    EFritekstFeil.MANGLER_TEKST,
+  );
+  const [erKnappTrykketPå, settKnappTrykketPå] = useState<boolean>(false);
 
-  const i18nInnhold = {
-    counterTooMuch: hentI18nInnhold(true),
-    counterLeft: hentI18nInnhold(false),
+  const fritekstfeltFeilmeldinger = {
+    [EFritekstFeil.MANGLER_TEKST]: tekster.fritekstfeltFeilmeldingManglerTekst,
+    [EFritekstFeil.HAR_SPESIAL_TEGN]:
+      tekster.fritekstfeltFeilmeldingSpesialTegn,
+    [EFritekstFeil.MINDRE_ENN_TI_TEGN]: tekster.fritekstfeltFeilmeldingMinTegn,
   };
 
-  function hentI18nInnhold(tegnIgjen: boolean) {
-    switch (språk) {
-      case 'nb':
-        return tegnIgjen ? 'tegn igjen' : 'tegn for mye';
-      case 'nn':
-        return tegnIgjen ? 'teikn igjen' : 'teikn for mykje';
-      case 'en':
-        return tegnIgjen ? 'characters left' : 'characters too many';
+  useEffect(() => {
+    if (actionData && actionData.text === RESPONSE_STATUS_OK) {
+      settEndringsmeldingMottattDato(actionData.mottattDato);
```

Comment:
Hvis actionData er av type `Response` må du ha `const data = await actionData.json()` for å pakke ut responsverdien når svaret kommer fra backend. 

Tenker man også bør ha en en state ala `feilmeldingFraServer` dersom responsen ikke er "OK" som vises under knappen: 
```Noe gikk galt ved innsending.```

Reply:
Jeg som ikke skjønner actionData - denne blir pakket ut gjennom actiondata, så ingenting å endre på her.